ejb3/src/main/java/org/jboss/as/ejb3/security/JaccInterceptor.java

```diff
@@ -157,7 +159,15 @@ protected MethodInterfaceType getMethodInterfaceType(MethodIntf viewType) {
     public static Principal[] getGrantedRoles(SecurityIdentity securityIdentity) {
         Set<String> roles = new HashSet<>();
 
-        securityIdentity.getRoles("ejb").forEach(roles::add);
-        return roles.stream().map((Function<String, Principal>) roleName -> (Principal) () -> roleName).toArray(Principal[]::new);
+        for (String s : securityIdentity.getRoles("ejb")) {
+            roles.add(s);
+        }
+        List<Principal> list = new ArrayList<>();
+        Function<String, Principal> mapper = roleName -> (Principal) () -> roleName;
+        for (String role : roles) {
+            Principal principal = mapper.apply(role);
+            list.add(principal);
+        }
+        return list.toArray(new Principal[0]);
```

Comment:
This is a throw-away array allocation on what might, given the class name, be the hot request path.

Reply:
It's probably deleted by the JIT (see https://shipilev.net/blog/2016/arrays-wisdom-ancients/ ), but I can hoist this to a constant anyway.

Reply:
Fixed.